Failed to execute gitui binary file downloaded from the release page

**Describe the bug**
A clear and concise description of what the bug is.

I downloaded gitui from https://github.com/extrawurst/gitui/releases/download/v0.10.0/gitui-linux-musl.tar.gz
extract and execute `./gitui`, failed with:
~~~
$ ./gitui
Failed to execute process './gitui'. Reason:
The file './gitui' does not exist or could not be executed.
~~~
And the ./gitui has the right permission.

FYI:
~~~
$ ldd ./gitui
linux-vdso.so.1 (0x00007ffe637e1000)
libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f706f942000)
libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f706f93d000)
libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f706f935000)
libc.so.6 => /usr/lib/libc.so.6 (0x00007f706f76c000)
/lib/ld64.so.1 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f706f9b0000)
~~~

**To Reproduce**
Steps to reproduce the behavior:
1. download https://github.com/extrawurst/gitui/releases/download/v0.10.0/gitui-linux-musl.tar.gz
2. extract it
3. execute it

**Expected behavior**
A clear and concise description of what you expected to happen.

Successfully working.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Context (please complete the following information):**
 - OS/Distro + Version: [5.7.17-2-MANJARO]
 - GitUI Version [v0.10.0]
 - Rust version: [rustc 1.45.0 (5c1f21c3b 2020-07-13)]

**Additional context**
Add any other context about the problem here.

Any hints on building from source?
```
> git config --get remote.origin.url
git@github.com:extrawurst/gitui.git

> git pull
Already up to date.

> cargo version
cargo 1.43.0

> rustc --version
rustc 1.43.0

> cargo install --path "."
  Installing gitui v0.10.0 (/home/yz/work/src/CODE/gitui)
    Updating crates.io index
   Compiling gitui v0.10.0 (/home/yz/work/src/CODE/gitui)
error[E0658]: `if` is not allowed in a `const fn`
   --> src/cmdbar.rs:128:9
    |
128 | /         if self.expandable && self.expanded {
129 | |             self.lines
130 | |         } else {
131 | |             1_u16
132 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `match` is not allowed in a `const fn`
  --> src/components/diff.rs:42:9
   |
42 | /         match self {
43 | |             Self::Single(start) | Self::Multiple(start, _) => *start,
44 | |         }
   | |_________^
   |
   = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: or-pattern is not allowed in a `const fn`
  --> src/components/diff.rs:43:13
   |
43 |             Self::Single(start) | Self::Multiple(start, _) => *start,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `match` is not allowed in a `const fn`
  --> src/components/diff.rs:48:9
   |
48 | /         match self {
49 | |             Self::Single(end) | Self::Multiple(_, end) => *end,
50 | |         }
   | |_________^
   |
   = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: or-pattern is not allowed in a `const fn`
  --> src/components/diff.rs:49:13
   |
49 |             Self::Single(end) | Self::Multiple(_, end) => *end,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `if` is not allowed in a `const fn`
   --> src/components/diff.rs:464:9
    |
464 | /         if hunk_min <= min && hunk_max >= max {
465 | |             return true;
466 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `if` is not allowed in a `const fn`
   --> src/components/diff.rs:469:9
    |
469 | /         if (hunk_min >= min && hunk_min <= max)
470 | |             || (hunk_max >= min && hunk_max <= max)
471 | |         {
472 | |             return true;
473 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `match` is not allowed in a `const fn`
   --> src/components/filetree.rs:209:9
    |
209 | /         match item_type {
210 | |             StatusItemType::Modified => 'M',
211 | |             StatusItemType::New => '+',
212 | |             StatusItemType::Deleted => '-',
213 | |             StatusItemType::Renamed => 'R',
214 | |             StatusItemType::Typechange => ' ',
215 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `if` is not allowed in a `const fn`
   --> src/ui/style.rs:128:9
    |
128 | /         if selected {
129 | |             style.bg(self.selection_bg)
130 | |         } else {
131 | |             style
132 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0658`.
error: failed to compile `gitui v0.10.0 (/home/yz/work/src/CODE/gitui)`, intermediate artifacts can be found at `/home/yz/work/src/CODE/gitui/target`

Caused by:
  could not compile `gitui`.

To learn more, run the command again with --verbose.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -72,6 +72,8 @@ jobs:
     - name: Build Debug
       run: |
         cargo build --target=x86_64-unknown-linux-musl
+        ls -lisa ./target/x86_64-unknown-linux-musl/debug
+        ldd ./target/x86_64-unknown-linux-musl/debug/gitui
         ./target/x86_64-unknown-linux-musl/debug/gitui --version
     - name: Build Release
       run: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -71,14 +71,11 @@ jobs:
         sudo apt-get -qq install musl-tools
     - name: Build Debug
       run: |
-        cargo build --target=x86_64-unknown-linux-musl
-        ls -lisa ./target/x86_64-unknown-linux-musl/debug
-        ldd ./target/x86_64-unknown-linux-musl/debug/gitui
+        make build-linux-musl-debug
         ./target/x86_64-unknown-linux-musl/debug/gitui --version
     - name: Build Release
       run: |
-        cargo build --release --target=x86_64-unknown-linux-musl
-        ldd ./target/x86_64-unknown-linux-musl/release/gitui
+        make build-linux-musl-release
         ./target/x86_64-unknown-linux-musl/release/gitui --version
 
   rustfmt:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -70,10 +70,14 @@ jobs:
       run: |
         sudo apt-get -qq install musl-tools
     - name: Build Debug
-      run: cargo build --target=x86_64-unknown-linux-musl
+      run: |
+        cargo build --target=x86_64-unknown-linux-musl
+        ./target/x86_64-unknown-linux-musl/debug/gitui --version
     - name: Build Release
       run: |
         cargo build --release --target=x86_64-unknown-linux-musl
+        ldd ./target/x86_64-unknown-linux-musl/release/gitui
+        ./target/x86_64-unknown-linux-musl/release/gitui --version
 
   rustfmt:
     name: Rustfmt

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.10.1] - 2020-09-01
+
+### Fixed
+- static linux binaries broke due to new clipboard feature which is disabled on linux for now ([#259](https://github.com/extrawurst/gitui/issues/259))
+
 ## [0.10.0] - 2020-08-29
 
 ### Added

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,7 +40,7 @@ serde = "1.0"
 anyhow = "1.0.32"
 unicode-width = "0.1"
 textwrap = "0.12"
-clipboard = "0.5"
+clipboard = { version = "0.5", optional = true }
 
 [target.'cfg(not(windows))'.dependencies]
 pprof = { version = "0.3", features = ["flamegraph"], optional = true }
@@ -49,7 +49,7 @@ pprof = { version = "0.3", features = ["flamegraph"], optional = true }
 maintenance = { status = "actively-developed" }
 
 [features]
-default=[]
+default=["clipboard"]
 timing=["scopetime/enabled"]
 
 [workspace]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "gitui"
-version = "0.10.0"
+version = "0.10.1"
 authors = ["Stephan Dilly <dilly.stephan@gmail.com>"]
 description = "blazing fast terminal-ui for git"
 edition = "2018"

--- a/Makefile
+++ b/Makefile
@@ -20,11 +20,16 @@ release-win: build-release
 	mkdir -p release
 	tar -C ./target/release/ -czvf ./release/gitui-win.tar.gz ./gitui.exe
 
-release-linux-musl: 
-	cargo build --release --target=x86_64-unknown-linux-musl
+release-linux-musl: build-linux-musl-release
 	strip target/x86_64-unknown-linux-musl/release/gitui
 	mkdir -p release
 	tar -C ./target/x86_64-unknown-linux-musl/release/ -czvf ./release/gitui-linux-musl.tar.gz ./gitui
+
+build-linux-musl-debug:
+	cargo build --target=x86_64-unknown-linux-musl --no-default-features
+
+build-linux-musl-release:
+	cargo build --release --target=x86_64-unknown-linux-musl --no-default-features
 
 test:
 	cargo test --workspace

--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -18,3 +18,13 @@ pub fn copy_string(string: String) -> Result<()> {
 pub fn copy_string(_string: String) -> Result<()> {
     Ok(())
 }
+
+#[cfg(feature = "clipboard")]
+pub fn is_supported() -> bool {
+    true
+}
+
+#[cfg(not(feature = "clipboard"))]
+pub fn is_supported() -> bool {
+    false
+}

--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -1,0 +1,20 @@
+use anyhow::Result;
+#[cfg(feature = "clipboard")]
+use clipboard::{ClipboardContext, ClipboardProvider};
+
+#[cfg(feature = "clipboard")]
+pub fn copy_string(string: String) -> Result<()> {
+    use anyhow::anyhow;
+
+    let mut ctx: ClipboardContext = ClipboardProvider::new()
+        .map_err(|_| anyhow!("failed to get access to clipboard"))?;
+    ctx.set_contents(string)
+        .map_err(|_| anyhow!("failed to set clipboard contents"))?;
+
+    Ok(())
+}
+
+#[cfg(not(feature = "clipboard"))]
+pub fn copy_string(_string: String) -> Result<()> {
+    Ok(())
+}

--- a/src/components/diff.rs
+++ b/src/components/diff.rs
@@ -604,11 +604,13 @@ impl Component for DiffComponent {
             self.focused,
         ));
 
-        out.push(CommandInfo::new(
-            strings::commands::copy(&self.key_config),
-            true,
-            self.focused,
-        ));
+        if crate::clipboard::is_supported() {
+            out.push(CommandInfo::new(
+                strings::commands::copy(&self.key_config),
+                true,
+                self.focused,
+            ));
+        }
 
         out.push(
             CommandInfo::new(
@@ -688,7 +690,9 @@ impl Component for DiffComponent {
                         }
                     }
                     Ok(true)
-                } else if e == self.key_config.copy {
+                } else if e == self.key_config.copy
+                    && crate::clipboard::is_supported()
+                {
                     self.copy_selection()?;
                     Ok(true)
                 } else {

--- a/src/components/diff.rs
+++ b/src/components/diff.rs
@@ -11,8 +11,6 @@ use crate::{
 use anyhow::Result;
 use asyncgit::{hash, sync, DiffLine, DiffLineType, FileDiff, CWD};
 use bytesize::ByteSize;
-#[cfg(feature = "clipboard")]
-use clipboard::{ClipboardContext, ClipboardProvider};
 use crossterm::event::Event;
 use std::{borrow::Cow, cell::Cell, cmp, path::Path};
 use tui::{
@@ -244,26 +242,6 @@ impl DiffComponent {
         Ok(())
     }
 
-    #[cfg(feature = "clipboard")]
-    fn copy_string(string: String) -> Result<()> {
-        use anyhow::anyhow;
-
-        let mut ctx: ClipboardContext = ClipboardProvider::new()
-            .map_err(|_| {
-                anyhow!("failed to get access to clipboard")
-            })?;
-        ctx.set_contents(string).map_err(|_| {
-            anyhow!("failed to set clipboard contents")
-        })?;
-
-        Ok(())
-    }
-
-    #[cfg(not(feature = "clipboard"))]
-    fn copy_string(_string: String) -> Result<()> {
-        Ok(())
-    }
-
     fn copy_selection(&self) -> Result<()> {
         if let Some(diff) = &self.diff {
             let lines_to_copy: Vec<&str> = diff
@@ -289,7 +267,9 @@ impl DiffComponent {
             try_or_popup!(
                 self,
                 "copy to clipboard error:",
-                Self::copy_string(lines_to_copy.join("\n"))
+                crate::clipboard::copy_string(
+                    lines_to_copy.join("\n")
+                )
             );
         }
 

--- a/src/components/diff.rs
+++ b/src/components/diff.rs
@@ -8,8 +8,10 @@ use crate::{
     strings, try_or_popup,
     ui::{self, calc_scroll_top, style::SharedTheme},
 };
+use anyhow::Result;
 use asyncgit::{hash, sync, DiffLine, DiffLineType, FileDiff, CWD};
 use bytesize::ByteSize;
+#[cfg(feature = "clipboard")]
 use clipboard::{ClipboardContext, ClipboardProvider};
 use crossterm::event::Event;
 use std::{borrow::Cow, cell::Cell, cmp, path::Path};
@@ -20,8 +22,6 @@ use tui::{
     widgets::{Block, Borders, Paragraph, Text},
     Frame,
 };
-
-use anyhow::{anyhow, Result};
 
 #[derive(Default)]
 struct Current {
@@ -244,7 +244,10 @@ impl DiffComponent {
         Ok(())
     }
 
+    #[cfg(feature = "clipboard")]
     fn copy_string(string: String) -> Result<()> {
+        use anyhow::anyhow;
+
         let mut ctx: ClipboardContext = ClipboardProvider::new()
             .map_err(|_| {
                 anyhow!("failed to get access to clipboard")
@@ -253,6 +256,11 @@ impl DiffComponent {
             anyhow!("failed to set clipboard contents")
         })?;
 
+        Ok(())
+    }
+
+    #[cfg(not(feature = "clipboard"))]
+    fn copy_string(_string: String) -> Result<()> {
         Ok(())
     }
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -10,6 +10,7 @@
 #![warn(clippy::missing_const_for_fn)]
 
 mod app;
+mod clipboard;
 mod cmdbar;
 mod components;
 mod input;

